src/bundle/Resources/public/js/scripts/udw/select.location.js

```diff
@@ -0,0 +1,136 @@
+    const token = doc.querySelector('meta[name="CSRF-Token"]').content;
+    const siteaccess = doc.querySelector('meta[name="SiteAccess"]').content;
+    const findLocationsByIdList = (idList, callback) => {
+        const body = JSON.stringify({
+            ViewInput: {
+                identifier: `udw-locations-by-path-string-${idList.join('-')}`,
+                public: false,
+                LocationQuery: {
+                    Criteria: {},
+                    FacetBuilders: {},
+                    SortClauses: { SectionIdentifier: 'ascending' },
+                    Filter: { LocationIdCriterion: idList.join(',') },
+                    limit: 50,
+                    offset: 0,
+                },
+            },
+        });
+        const request = new Request('/api/ezp/v2/views', {
+            method: 'POST',
+            headers: {
+                Accept: 'application/vnd.ez.api.View+json; version=1.1',
+                'Content-Type': 'application/vnd.ez.api.ViewInput+json; version=1.1',
+                'X-Requested-With': 'XMLHttpRequest',
+                'X-Siteaccess': siteaccess,
+                'X-CSRF-Token': token,
+            },
+            body,
+            mode: 'same-origin',
+            credentials: 'same-origin',
+        });
+        const errorMessage = Translator.trans(
+            /*@Desc("Cannot find children locations with given id - %idList%")*/ 'select_location.error',
+            { idList: idList.join(',') },
+            'universal_discovery_widget'
+        );
+
+        fetch(request)
+            .then(eZ.helpers.request.getJsonFromResponse)
+            .then(callback)
+            .catch(() => eZ.helpers.notification.showErrorNotification(errorMessage));
+    };
+    const removeRootFromPathString = (pathString) => {
+        const pathArray = pathString.split('/').filter((val) => val);
+
+        return pathArray.splice(1, pathArray.length - 1);
+    };
+    const buildBreadcrumbsString = (viewData) => {
+        const searchHitList = viewData.View.Result.searchHits.searchHit;
+
+        return searchHitList.map((searchHit) => searchHit.value.Location.ContentInfo.Content.Name).join(' / ');
+    };
+    const toggleVisibility = (btn, isLocationSelected) => {
+        const contentBreadcrumbsWrapper = doc.querySelector(btn.dataset.contentBreadcrumbsSelector);
+
+        btn.hidden = isLocationSelected;
+
+        if (contentBreadcrumbsWrapper) {
+            contentBreadcrumbsWrapper.hidden = !isLocationSelected;
+        }
+    };
+    const updateBreadcrumbsState = (btn, pathString) => {
+        const pathStringInput = doc.querySelector(btn.dataset.locationPathInputSelector);
+        const contentBreadcrumbs = doc.querySelector(btn.dataset.contentBreadcrumbsSelector);
+        const contentBreadcrumbsContainer = contentBreadcrumbs.querySelector('.ez-filters__breadcrumbs');
+        const contentBreadcrumbsSpinner = contentBreadcrumbs.querySelector('.ez-filters__breadcrumbs-spinner');
```

Comment:
cc @inakijv can you check the final result of showing the loading spinner while fetching data about the location path? Please, tell us whether you like it.